src/core/src/lib/templates/field.type.ts

```diff
@@ -15,14 +15,14 @@ export abstract class FieldType<F extends FormlyFieldConfig = FormlyFieldConfig>
   set form(form) { console.warn(`NgxFormly: passing 'form' input to '${this.constructor.name}' component is not required anymore, you may remove it!`); }
 
   @Input()
-  get options(): F['options'] { return this.field.options; }
+  get options() { return this.field.options; }
   set options(options: F['options']) { console.warn(`NgxFormly: passing 'options' input to '${this.constructor.name}' component is not required anymore, you may remove it!`); }
 
   get key() { return this.field.key; }
 
-  get formControl(): F['formControl'] { return this.field.formControl; }
+  get formControl() { return this.field.formControl; }
 
-  get to(): F['templateOptions'] { return this.field.templateOptions; }
+  get to() { return this.field.templateOptions || {}; }
```

Comment:
By adding  it I assure that object is there always. Otherwise TypeScript will recognise it as  an optional field and will require bunch of `if` statements all around the code, or `this.to!.something`, which is equally as ugly.

Do you  see any other option?

Reply:
keep it then :)